arcgis-ios-sdk-samples/Utility Network/Configure subnetwork trace/ConfigureSubnetworkTraceViewController.swift

```diff
@@ -0,0 +1,391 @@
+                self.selectedValue = textfield.text
+            }
+        })
+        // Add cancel button.
+        alert.addAction(UIAlertAction(title: "Cancel", style: .cancel))
+        
+        // Present the alert.
+        self.present(alert, animated: true, completion: nil)
+    }
+    
+    func addCondition() {
+        if configuration == nil {
+            configuration = AGSUtilityTraceConfiguration()
+        }
+        if configuration?.traversability == nil {
+            configuration?.traversability = AGSUtilityTraversability()
+        }
+        // NOTE: You may also create a UtilityCategoryComparison with UtilityNetworkDefinition.Categories and UtilityCategoryComparisonOperator.
+        if selectedAttribute != nil {
+            // If the value is a coded value.
+            if let codedValue = selectedValue as? AGSCodedValue, selectedAttribute?.domain is AGSCodedValueDomain {
+                selectedValue = convertToDataType(otherValue: codedValue.code!, dataType: selectedAttribute!.dataType)
+            } else {
+                selectedValue = convertToDataType(otherValue: selectedValue!, dataType: selectedAttribute!.dataType)
+            }
+            // NOTE: You may also create a UtilityNetworkAttributeComparison with another NetworkAttribute.
+            var expression: AGSUtilityTraceConditionalExpression?
+            expression = AGSUtilityNetworkAttributeComparison(networkAttribute: selectedAttribute!, comparisonOperator: selectedComparison!, value: selectedValue!)
+            if let otherExpression = configuration?.traversability?.barriers as? AGSUtilityTraceConditionalExpression {
+                // NOTE: You may also combine expressions with UtilityTraceAndCondition
+                expression = AGSUtilityTraceOrCondition(leftExpression: otherExpression, rightExpression: expression!)
+            }
+            // Update the list of barrier conditions.
+            configuration?.traversability?.barriers = expression
+            let expressionString = expressionToString(expression: expression!)
+            textView?.text += "\n \(expressionString!))"
+        }
+    }
+    
+    func reset() {
+        // Reset the barrier condition to the initial value.
+        let traceConfiguration = configuration
+        traceConfiguration?.traversability?.barriers = initialExpression
+        textView?.text = expressionToString(expression: initialExpression!)
+        // Reset the selected values.
+        selectedAttribute = nil
+        selectedComparison = nil
+        selectedValue = nil
+        //Reset the labels.
+        attributeLabel.text = nil
+        comparisonLabel.text = nil
+        valueLabel.text = nil
+    }
+    
+    func trace() {
+        if utilityNetwork == nil || startingLocation == nil {
+            presentAlert(title: "Error", message: "Could not trace utility network.")
+            return
```

Comment:
```suggestion
```
This return seem to be redundant.

Also, could the availability of utility network be decided during its `load`?

Reply:
Availability is determined during its `load` but this is just an extra precaution as also seen in other SDK implementations. It can be removed if we deem it redundant.